CommuniTEAM/CommuniTEA
# TEA-69-Apply-authentication-to-create-teas

### In this PR, I'm adding/changing the following:

Applied authorization to create teas function. 
Modified SQL for GetCity to generate UUID when given state abbreviation and city name params. 
Modified create user function to take city name and state abbreviation inputs to generate location UUID instead of manually inputting location UUID.

### Screenshots / Code Snippets:

screenshot of creating a new user:

<img width="938" alt="image" src="https://github.com/CommuniTEAM/CommuniTEA/assets/134474081/72a9500a-7bae-4d9d-8057-2a15caf8841a">

screenshot of creating a new tea:

<img width="1067" alt="image" src="https://github.com/CommuniTEAM/CommuniTEA/assets/134474081/017a1497-7e06-4dd8-a7f1-96339b864c15">


### Please do the following to test my code:

1. Run docker.
2. Logout and create a tea. (Should have permission denied)
3. Create a city in beekeeper if not already done.
4. Create a new user.
5. Create a new tea. (below is a json example)
```
{
  "brew_temp ": 175,
  "brew_time ": "5 minutes",
  "description": "Premium Earl Grey tea.",
  "img_url ": "https://images.unsplash.com/photo-1605618826115-fb9e775cfb40?q=80&w=1779&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D",
  "name": "The Earl of Grey"
}
```

[Related Jira Ticket](https://communiteam.atlassian.net/browse/TEA-69)
